cpp/include/raft/stats/dispersion.cuh

```diff
@@ -51,10 +53,51 @@ DataT dispersion(const DataT* centroids,
                  IdxT dim,
                  cudaStream_t stream)
 {
-  return detail::dispersion(
+  return detail::dispersion<DataT, IdxT, TPB>(
     centroids, clusterSizes, globalCentroid, nClusters, nPoints, dim, stream);
 }
 
+/**
+ * @brief Compute cluster dispersion metric. This is very useful for
+ * automatically finding the 'k' (in kmeans) that improves this metric.
+ * @tparam value_t data type
+ * @tparam idx_t index type
+ * @param[in]  handle the raft handle
+ * @param[in]  centroids the cluster centroids. This is assumed to be row-major
+ *   and of dimension (n_clusters x dim)
+ * @param[in]  cluster_sizes number of points in the dataset which belong to each
+ *   cluster. This is of length n_clusters
+ * @param[out] global_centroid compute the global weighted centroid of all cluster
+ *   centroids. This is of length dim. Use std::nullopt to not return it.
+ * @param[in]  n_points number of points in the dataset
+ * @return the cluster dispersion value
+ */
+template <typename value_t, typename idx_t>
+value_t dispersion(const raft::handle_t& handle,
```

Comment:
Now might be a good time to rename this to cluster_dispersion (there are many types of dispersion measures, and also many types of cluster dispersion measures). We probably also want to spell out the math in the description so a user knows when to use this (the use of the term "centroids" already reduces the usefulness of this with an algorithm that assume gaussian distributed data- such as most density-based methods).